### ISSUE TITLE
crash with TOC header in html

### ISSUE THREAD

**Issue by Fighter777** [mikehaertl/phpwkhtmltopdf#219]
hello, phpwkhtmltopdf don't convert my html header TOC in temporaly file html and send the html escaped to wkhtmltopdf

`Failed without error message: C: && cd "C:\inetpub\src\Controllers\..\..\..\resources\bin\32" && wkhtmltopdf_12.4_MSVC.exe --orientation "landscape" --page-size "A5" --disable-smart-shrinking --margin-bottom "10" --margin-top "20" --margin-left "5" --margin-right "5" "C:\inetpub\temp\tmpAFC.tmp.html" --header-html "C:\inetpub\temp\tmpAEC.tmp.html" --header-line --footer-center "CONFIDENTIEL" --footer-left "[date] [time]" --footer-line --header-spacing "0.2" toc --header-html "< DOCTYPE html>						<html>							<head>								<meta http-equiv= Content-Type  content= text/html; charset=UTF-8  />								<link href= ../public/static/css/pdf/editeur.css?ref  rel= stylesheet >								<link href= ../public/static/css/generation_livret/model_livret.css?ref  rel= stylesheet >							</head>							<body>								<section>									<div class= header >									</div>								</section>							</body>						</html>" --header-line --toc-header-text "sommaire" --footer-center "CONFIDENTIEL" --footer-right "[page]/[toPage]" --footer-left "[date] [time]" --footer-line --header-spacing "0.2" "C:\inetpub\temp\tmpAFE.tmp.html" --header-line --header-html "C:\inetpub\temp\tmpAFD.tmp.html" --footer-right "[page]/[toPage]" --footer-center "CONFIDENTIEL" --footer-left "[date] [time]" --footer-line --header-spacing "0.2" "C:\inetpub\temp\tmpB0F.tmp.pdf`

that work with a url

**Comment by mikehaertl** [comment]
Hmm, can you do some debugging? I would focus on the block here:

https://github.com/mikehaertl/phpwkhtmltopdf/blob/master/src/Pdf.php#L305

Find out, why no new File is created in line 309.

**Comment by mikehaertl** [comment]
I think I found it. Can you try to add this line to `Pdf.php` before line 146:

```
public function addToc($options=array())
{
// Add this:
$options = $this->processOptions($options);
```

**Comment by mikehaertl** [comment]
@Fighter777 The above fix is available on master. Could you help testing please?

**Comment by Fighter777** [comment]
hello, i test that tomorrow ;)

**Comment by Fighter777** [comment]
its good for me, thx you 👍

**Comment by mikehaertl** [comment]
Thanks. The fix is included in the new 2.2.1 release.
